search/tasks.py

```diff
@@ -35,3 +74,127 @@ def update_document_with_partial(doc_id, partial_data):
 def increment_document_integer_field(doc_id, field_name, incr_amount):
     """Task that makes a request to increment some integer field in an ES document"""
     api.increment_document_integer_field(doc_id, field_name, incr_amount)
+
+
+@app.task(autoretry_for=(RetryException, ), retry_backoff=True, rate_limit='600/m')
+def index_post(api_username, post_id):
+    """
+    Index a post and its comments
+
+    Args:
+        api_username (str): The API username
+        post_id (str): The post string
+    """
+    with wrap_retry_exception(PrawcoreException, PRAWException):
+        api.index_post(api_username, post_id)
+
+
+@app.task(bind=True, autoretry_for=(RetryException, ), retry_backoff=True, rate_limit='600/m')
+def index_channel(self, api_username, channel_name):
+    """
+    Index the channel
+
+    Args:
+        api_username (str): API username
+        channel_name (str): The name of the channel
+    """
+    with wrap_retry_exception(PrawcoreException, PRAWException):
+        from channels.api import Api
+
+        client = Api(User.objects.get(username=api_username))
+        channel = client.get_channel(channel_name)
+        post_ids = [post.id for post in channel.new()]
+
+    raise self.replace(
+        group(
+            index_post.si(api_username=api_username, post_id=post_id) for post_id in post_ids
+        )
+    )
+
+
+@app.task(bind=True)
+def start_recreate_index(self, api_username):
+    """
+    Wipe and recreate index and mapping, and index all items.
+
+    Args:
+        api_username (str): The reddit username to use when crawling data
+    """
+    from channels.api import Api
+
+    user = User.objects.get(username=api_username)
```

Comment:
We were planning on having a single admin user for crawling & indexing, right? Would it make more sense to define a setting for the crawler/indexer username and use that in all of these tasks rather than accepting a parameter?

Reply:
Makes sense to me